tests/integration/schema_registry/test_api_client.py

```diff
@@ -176,9 +176,9 @@ def test_api_get_register_schema_invalid(kafka_cluster, load_file):
 
     with pytest.raises(SchemaRegistryError, match="Invalid schema") as e:
         sr.lookup_schema(subject, schema2)
-    # Not as documented but the caused by is correct.
-    assert e.value.http_status_code == 500
-    assert e.value.error_code == 500
+
+    assert e.value.http_status_code == 422
+    assert e.value.error_code == 42201
```

Comment:
42201 is subject not found error, shouldn't this be 42202?

Reply:
this is what's returned when the test passes. That error code 42201 is for an invalid schema [stated here](https://docs.confluent.io/platform/current/schema-registry/develop/api.html) 42202 is for invalid version. In this case it's invalid schema because of the `"type" : "array"`, attribute in the avsc

Reply:
I think that the [documentation stayed the same](https://docs.confluent.io/platform/6.0.0/schema-registry/develop/api.html#post--subjects-(string-%20subject)-versions) but in version 7.0.2 they have fixed it and started returning 422 instead of internal server error.